### PR TITLE
Add additional ML cluster settings

### LINKED ISSUE #4237: [DOC] Need to add More setting values in the MLCommons doc
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 
We need to add documentation about few more settings in the [MLCommons Cluster Settings page](https://opensearch.org/docs/latest/ml-commons-plugin/cluster-settings/)

Setting name: `plugins.ml_commons.enable_inhouse_python_model`

Description:  From 2.7, we are introducing python based in house model in Opensearch cluster. [Metrics Correlation] algorithm is the first initiative. To run python based in house models, we need to enable (change the value to `true`] this setting first. 

Default Value: false
Value Range: [false, true]

Setting name: `plugins.ml_commons.allow_registering_model_via_url`

Description:  If we want to register any model via url in the opensearch cluster, we need to enable this settings first. By default, we don't allow any model registration through URL.

Default Value: false
Value Range: [false, true]

Setting name: `plugins.ml_commons.allow_registering_model_via_local_file`

Description:  If we want to register any model from local file in the opensearch cluster, we need to enable this settings first. By default, we don't allow any model registration from local file upload/register

Default Value: false
Value Range: [false, true]

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
Fixes #4237 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _ml-commons-plugin/cluster-settings.md [hunk 1]
```diff
@@ -120,24 +120,54 @@ plugins.ml_commons.max_upload_model_tasks_per_node: 10
...
+```
+
+### Values
+
+- Default value: false
+- Value range: [false, true]
+
+## Register models using local files
+
+This setting gives you the ability to register a model using a local file. By default, ML Commons only allows model registration through portable formats.
```
[the remark this answers is not in the document]
Comment (dhrubo-os) [reply _ml-commons-plugin/cluster-settings.md:155]: Can we also make the related changes [in this page](https://opensearch.org/docs/latest/ml-commons-plugin/model-serving-framework/)?

File: _ml-commons-plugin/cluster-settings.md [hunk 2]
```diff
@@ -103,14 +103,14 @@ plugins.ml_commons.monitoring_request_count: 100
 - Default value: `100`
 - Value range: [0, 10,000,000]
 
-## Upload model tasks per node
+## Register model tasks per node
 
-Controls how many upload model tasks can run in parallel on one node. If set to `0`, you cannot upload models to any node.
+Controls how many registered model tasks can run in parallel on one node. If set to `0`, you cannot register models to any node.
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/cluster-settings.md:108]:
```suggestion
Controls how many register model tasks can run in parallel on one node. If set to `0`, you cannot run register model tasks on any node.
```

File: _ml-commons-plugin/cluster-settings.md [hunk 3]
```diff
@@ -120,24 +120,54 @@ plugins.ml_commons.max_upload_model_tasks_per_node: 10
...
 ```
 
 ### Values 
 
 - Default value: `10`
 - Value range: [0, 10]
 
+## Register models using URLs
+
+This setting gives you the ability to register models using a URL. By default, ML Commons only allows model registration through portable formats.
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/cluster-settings.md:140]:
```suggestion
This setting gives you the ability to register models using a URL. By default, ML Commons only allows register pre-trained model from OpenSearch model repo, read this doc about pre-trained model https://opensearch.org/docs/latest/ml-commons-plugin/pretrained-models/
```

File: _ml-commons-plugin/cluster-settings.md [hunk 4]
```diff
@@ -246,3 +276,23 @@ plugins.ml_commons.model_auto_redeploy_success_ratio: 0.8
 
 - Default value: 0.8
 - Value range: [0, 1]
+
+## Run Python-based models
+
+When set to `true`, this setting enables the ability to run Python-based models, such as [Metrics correlation]({{site.url}}{{site.baseurl}}/ml-commons-plugin/algorithms/#metrics-correlation).
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/cluster-settings.md:282]: Is this accurate ? @dhrubo-os 

I think this setting only controls whether user can run the model built by OpenSearch team.

Comment (dhrubo-os) [reply _ml-commons-plugin/cluster-settings.md:282]: Good point @ylwu-amzn. May be we can update it to: `this setting enables the ability to run Python-based models built by Opensearch team`

File: _ml-commons-plugin/cluster-settings.md (hunk 1 above)
Comment (Naarcha-AWS) [inline _ml-commons-plugin/cluster-settings.md:155]:
```suggestion
This setting gives you the ability to register a model using a local file. By default, ML Commons only allows registration of [pretrained]({{site.url}}{{site.baseurl}}//ml-commons-plugin/pretrained-models/) models from the OpenSearch model repository.
```

File: _ml-commons-plugin/cluster-settings.md (hunk 4 above)
Comment (Naarcha-AWS) [inline _ml-commons-plugin/cluster-settings.md:282]:
```suggestion
When set to `true`, this setting enables the ability to run Python-based models supported by OpenSearch, such as [Metrics correlation]({{site.url}}{{site.baseurl}}/ml-commons-plugin/algorithms/#metrics-correlation).
```

Comment (dhrubo-os) [reply _ml-commons-plugin/cluster-settings.md]: I'm not sure if supported will be the right word here. 

For example: these [python based pre-trained models](https://opensearch.org/docs/latest/ml-commons-plugin/pretrained-models/), we didn't develop those, we just traced it and we are supporting this.

But `enable_inhouse_python_model` is only applicable if any model is developed in python by opensearch team. I hope this clarifies the distinction here.

File: _ml-commons-plugin/cluster-settings.md [hunk 5]
```diff
@@ -103,14 +103,14 @@ plugins.ml_commons.monitoring_request_count: 100
 - Default value: `100`
 - Value range: [0, 10,000,000]
 
-## Upload model tasks per node
+## Register model tasks per node
 
-Controls how many upload model tasks can run in parallel on one node. If set to `0`, you cannot upload models to any node.
+Controls how many register model tasks can run in parallel on one node. If set to `0`, you cannot run register model tasks on any node.
```
Comment (cwillum) [inline _ml-commons-plugin/cluster-settings.md:108]: Just wondering. What do you think about hyphens for these compounds?
register-model tasks.
deply-model tasks.

**Review by cwillum — APPROVED** [review]
Looks good.

File: _ml-commons-plugin/cluster-settings.md (hunk 5 above)
Comment (Naarcha-AWS) [reply _ml-commons-plugin/cluster-settings.md:108]: Let me think about that one.

**Pull request merged**